util/build_docs.py

```diff
@@ -191,9 +192,17 @@ def generate_hardware_blocks():
 
 def generate_testplans():
     for testplan in config["testplan_definitions"]:
+        # Split the filename into filename and tags, if provided.
+        split = str(testplan).split(":")
+        filename = split[0]
+        if split[1:]:
+            tags = "_".join(split[1:])
+        else:
+            tags = ""
         plan = Testplan.Testplan(SRCTREE_TOP.joinpath(testplan),
                                  repo_top=SRCTREE_TOP)
-        plan_path = config["outdir-generated"].joinpath(testplan + '.testplan')
+        plan_path = config["outdir-generated"].joinpath(
+            f"{filename}.{tags}_testplan")
```

Comment:
Nit (if you're pushing anyway...): I'd probably assign the basename to a local variable on the line above. It will still be two lines, but you get to give something a name and the formatting looks less weird. Something like this, maybe?
```suggestion
        plan_filename = f"{filename}.{tags}_testplan"
        plan_path = config["outdir-generated"].joinpath(plan_filename)
```